cameo/flux_analysis/simulation.py

```diff
@@ -74,59 +73,61 @@ def fba(model, objective=None, reactions=None, *args, **kwargs):
         Contains the result of the linear solver.
 
     """
-    with TimeMachine() as tm:
+    with model:
         if objective is not None:
-            tm(do=partial(setattr, model, 'objective', objective),
-               undo=partial(setattr, model, 'objective', model.objective))
-        solution = model.solve()
+            model.objective = objective
+        model.solver.optimize()
+        if model.solver.status != 'optimal':
```

Comment:
I think it should raise an exception according to the status: infeasible, unbounded, etc.